app/controllers/concerns/json_api_pagination.rb

```diff
@@ -40,4 +43,9 @@ def pagination_url(page_number)
 
     url_for(only_path: false, params: request.query_parameters.merge(page: page_number))
   end
+
+  # User-specified value, or DEFAULT_PER_PAGE, capped at MAX_PER_PAGE
+  def final_per_page_value
+    (params[:per_page] || DEFAULT_PER_PAGE).to_i.clamp(1, MAX_PER_PAGE)
+  end
```

Comment:
Great! A method like this will always allow us to implement something different here or override the method in a controller without messing with constants! :+1: